src/main/include/subsystems/Intake.hpp

```diff
@@ -2,15 +2,110 @@
 
 #pragma once
 
+#include <frc/AnalogInput.h>
+#include <frc/Solenoid.h>
+#include <rev/CANSparkMax.h>
+
 #include "Constants.hpp"
 #include "communications/PublishNode.hpp"
 #include "subsystems/SubsystemBase.hpp"
 
 namespace frc3512 {
 
+enum class State {
+    kIdle,
+    // exit state when intake runs
+    kDropBalls,
+    // conveyor and intake are on, reverse motor so that conveyor drops balls to
+    // bottom
+    kRaiseBalls
+    // when hopper is full, run conveyor motor to bring balls back up
+};
+
 class Intake : public SubsystemBase, public PublishNode {
```

Comment:
I suggest adding a Doxygen comment above the class here that describes the hardware layout of the robot (e.g., intake can intake, outtake, and stow itself; intake feeds into conveyor; conveyor has proximity sensors on top and bottom to detect balls).